descwl_coadd/coadd.py

```diff
@@ -276,11 +412,186 @@ def make_coadd(
+
+    # can re-use the warper for each coadd type except the mfrac where we use
+    # linear
+    warp_config = afw_math.Warper.ConfigClass()
+    warp_config.warpingKernelName = DEFAULT_INTERP
+    warper = afw_math.Warper.fromConfig(warp_config)
+
+    warp_config = afw_math.Warper.ConfigClass()
+    warp_config.warpingKernelName = "bilinear"
+    mfrac_warper = afw_math.Warper.fromConfig(warp_config)
+
+    # will zip these with the exposures to warp and add
+    wcss = [coadd_wcs, coadd_wcs, coadd_psf_wcs, coadd_wcs]
+    bboxes = [coadd_bbox, coadd_bbox, coadd_psf_bbox, coadd_bbox]
+    warpers = [warper, warper, warper, mfrac_warper]
+
+    # PSF will generally have NO_DATA in warp as we tend to use the same psf
+    # stamp size for input and output psf and just zero out wherever there is
+    # no data
+
+    verifys = [True, True, False, True]
+
+    LOG.info('warping and adding exposures')
+
+    if isinstance(exp, DeferredDatasetHandle):
+        expobj = exp.get()
+    else:
+        expobj = exp
+
+    # TODO use exp.getId() when it arrives in weekly 44.  For now use an index
+    # 0::len(exps)
+    try:
+        exp_id = expobj.getId()
+    except AttributeError:
+        exp_id = -9999
+
+    bad_msk, maskfrac = get_bad_mask(expobj)
+
+    noise_exp, medvar = get_noise_exp(
+        exp=expobj, rng=rng, remove_poisson=remove_poisson,
+    )
+
+    exp_info = get_info_struct(1)
+    exp_info['exp_id'] = exp_id
+    exp_info['maskfrac'] = maskfrac
+    exp_info['weight'] = 1/medvar
+
+    try:
+        if maskfrac >= max_maskfrac:
+            raise HighMaskedFrac('high maskfrac')
+
+        mfrac_exp = make_mfrac_exp(mfrac_msk=bad_msk, exp=expobj)
+
+        if maskfrac > 0:
+            # images modified internally
+            interp_nocheck(exp=expobj, noise_exp=noise_exp, bad_msk=bad_msk)
+
+        psf_exp = get_psf_exp(
+            exp=expobj,
+            coadd_cen_skypos=coadd_cen_skypos,
+            var=medvar,
+        )
+        assert psf_exp.variance.array[0, 0] == noise_exp.variance.array[0, 0]
+
+        # we use this to check no edges made it into the coadd
+        add_boundary_bit(expobj)
+        add_boundary_bit(noise_exp)
+
+        # order must match wcss, bboxes, warpers, verifys
+        exps2add = [expobj, noise_exp, psf_exp, mfrac_exp]
+
+        # the verify checks boundary/NO_DATA Use an exception as an easy way to
+        # guarantee that if one fails to verify we fail
+        warps = _get_warps_for_exp(exps2add, wcss, bboxes, warpers, verifys)
+        warp, noise_warp, psf_warp, mfrac_warp = warps
+
+    except WarpBoundaryError as err:
+        LOG.info('%s', err)
+        exp_info['flags'] |= WARP_BOUNDARY
+        warp, noise_warp, psf_warp, mfrac_warp = [None] * 4
+    except HighMaskedFrac:
+        LOG.info(f'{exp_id} maskfrac {maskfrac} >= {max_maskfrac}')
+        exp_info['flags'] |= HIGH_MASKFRAC
+        warp, noise_warp, psf_warp, mfrac_warp = [None] * 4
```

Comment:
I think warps should be generated and persistent regardless of the mask fraction. The choice of whether to include them in the coadd or not should happen downstream when we assemble them into coadds.